src/sysext/sysext.c

```diff
@@ -309,8 +311,15 @@ static int mount_overlayfs(
                 separator = true;
         }
 
+        /* confexts are nosuid by default, even if they are exec */
+        if (arg_image_class == IMAGE_CONFEXT)
+                flags |= MS_NOSUID;
```

Comment:
please extend the image_class_info[] structure above with a new field `default_mount_options` or so, and then se MS_NOSUID|MS_NODEV in the field in IMAGE_CONFEXT and MS_NODEV for IMAGE_SYSEXT